### PR TITLE
adjust BatchResult to interface cleanup

### PR DESCRIPTION
**Pull request by dbu**
depends on https://github.com/php-http/httplug/pull/73

### CODE REVIEW

**Comment by dbu** [comment]
tests will fail until https://github.com/php-http/httplug/pull/73 is merged.

File: src/BatchResult.php [hunk 1]
```diff
@@ -117,29 +109,21 @@ public function getExceptions()
...
      * {@inheritDoc}
      */
-    public function hasExceptionFor(RequestInterface $request)
+    public function getExceptionFor(RequestInterface $request)
     {
-        return $this->exceptions->contains($request);
+        try {
+            return $this->exceptions[$request];
+        } catch (\UnexpectedValueException $e) {
+            throw new UnexpectedValueException('Request not found', $e->getCode(), $e);
```
Comment (sagikazarmark) [inline src/BatchResult.php]: Isn't this exception an SPL UnexpectedException as well?

Comment (dbu) [inline src/BatchResult.php]: (i only reordered this method, did not change any content)

seems to be the same yes. but the exception message will become more specific like this, which seems to be worth it.

Comment (sagikazarmark) [inline src/BatchResult.php]: I think I did this rethrow thing in order to hide the original Object not found message. But the question is: If it is the same, why isn't it used from the root namespace as well?

Comment (dbu) [inline src/BatchResult.php]: we have a `use UnexpectedValueException` in this file. probably against conventions, i can remove it and use `\UnexpectedValueException` everywhere. ok?

Comment (sagikazarmark) [inline src/BatchResult.php]: :+1:

**Pull request merged**